### ISSUE TITLE
Support Views in Flask-style decorators (RouteTableDef)

### ISSUE THREAD

**Issue by socketpair** [aio-libs/aiohttp#2472]
see https://github.com/aio-libs/aiohttp/pull/2471#issuecomment-342299246

**Comment by pfreixes** [comment]
I was taking a look at that and looks hard to give a 100% of consistency between decorators and views. Likely the best way to give support for decorating views is creating a new ad-hoc method called `view`,  see the following example: 

```python

routes = web.RouteTableDef()

@routes.view('/view')
class MyView(web.View):
    async def get(self):
        return await get_resp(self.request)

    async def post(self):
        return await post_resp(self.request)
```

Thoughts @asvetlov @socketpair?

**Comment by asvetlov** [comment]
Brilliant idea.
Let's add `web.view` function and `router.add_view()` too -- in this case everything should interconnect smoothly.

**Comment by sheb** [comment]
Is anyone working on this ?

**Comment by asvetlov** [comment]
AFAIK nobody.

**Comment by pfreixes** [comment]
On my TODO, but too many things these days. If you are keen on work on it,
go ahead

**Comment by sheb** [comment]
Ok I'll take care of it.
PR maybe this week-end

**Comment by asvetlov** [comment]
Fixed by #2611
